### ISSUE TITLE
std::out_of_range error in running demo cartographer_paper_deutsches_museum.bag

### ISSUE THREAD

**Issue by UditSinghParihar** [googlecartographer/cartographer_ros#1265]
I am trying to run cartographer on demo bag file, mentioned in the [tutorial]( https://google-cartographer-ros.readthedocs.io/en/latest/demos.html#id1), but it is giving following error:
1. `$ roslaunch cartographer_ros demo_backpack_2d.launch bag_filename:=${HOME}/Downloads/cartographer_paper_deutsches_museum.bag`

2. Error:
```
[ INFO] [1555409027.236835694, 1405076296.503967961]: I0416 15:33:47.000000 17593 ordered_multi_queue.cc:172] All sensor data for trajectory 0 is available starting at '635406730964746083'.
terminate called after throwing an instance of 'std::out_of_range'
  what():  map::at

```
3. Distro- `Ubuntu16.04, Kinetic`. My installation process was error free. So what could be the possible reason.

Thanks.

**Comment by pifon2a** [comment]
Are you using Cartographer provided with ROS?

**Comment by UditSinghParihar** [comment]
Yeah I am using Cartographer with ROS, with instructions from this [page](https://google-cartographer-ros.readthedocs.io/en/latest/compilation.html)

**Comment by UditSinghParihar** [comment]
Ok, I solved it. I was basically using `PCL1.9`, which can only be used with `c++14`. So I change 
`CMakeLists.txt` in `cartographer_ros` to be compiled with `c++14`. This was causing the error.  After using `PCL1.7` which does not require to be used with `c++14`, I was able to build cartographer without changing anything in the `CMakeLists.txt`.  
I think we could close this issue.  

Thanks
